### ISSUE TITLE
getrandom() unavailable on macOS

### ISSUE THREAD

**Issue by shreemaan-abhishek** [liweitianux/nanoid#1]
I tried to install `lua-nanoid` luarock on my system (M1 Mac) but it failed with the following error:

```
lua-nanoid 1.0.0-1 depends on lua >= 5.1 (5.1-1 provided by VM)
env MACOSX_DEPLOYMENT_TARGET=11.0 gcc -O2 -fPIC -I/opt/homebrew/opt/lua@5.1/include/lua5.1 -c nanoid_lua.c -o nanoid_lua.o -DNDEBUG
env MACOSX_DEPLOYMENT_TARGET=11.0 gcc -O2 -fPIC -I/opt/homebrew/opt/lua@5.1/include/lua5.1 -c nanoid.c -o nanoid.o -DNDEBUG
nanoid.c:90:9: error: call to undeclared function 'getrandom'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        getrandom(bytes, sizeof(bytes), 0);
        ^
1 error generated.

Error: Failed installing dependency: https://luarocks.org/lua-nanoid-1.0.0-1.rockspec - Build error: Failed compiling object nanoid.o
make: *** [deps] Error 1
```

**Comment by liweitianux** [comment]
Partially resolved by commit 5e7fecda157b878170c34d6c001637754e373f7d.

I'd like to use `arc4random(3)` for better performance and better portable with old BSDs and macOS versions.  However, Linux needs some workaround ...

For the moment, please try `luarocks install rockspec/lua-nanoid-master-1.rockspec`.

I'll update the LuaRocks submission later.

**Comment by shreemaan-abhishek** [comment]
@liweitianux, thank you so much. But to be used in apisix, we need to make sure the algo works on all platforms

**Comment by liweitianux** [comment]
Sure, so I'm still working on this issue.

**Comment by liweitianux** [comment]
Hi, sorry for the delay. I've been quite busy in recent weeks. I'll resume the work this weekend and hopefully resolve this issue.

**Comment by liweitianux** [comment]
Fixed in release v1.1.0.

Nonetheless, I'd like to refactor the API to improve the performance of using `/dev/urandom`.

**Comment by shreemaan-abhishek** [comment]
It still doesn't work on macos:

```
lua-nanoid 1.1.0-1 depends on lua >= 5.1 (5.1-1 provided by VM)
env MACOSX_DEPLOYMENT_TARGET=11.0 gcc -O2 -fPIC -I/opt/homebrew/opt/lua@5.1/include/lua5.1 -c nanoid_lua.c -o nanoid_lua.o -DNDEBUG
env MACOSX_DEPLOYMENT_TARGET=11.0 gcc -O2 -fPIC -I/opt/homebrew/opt/lua@5.1/include/lua5.1 -c nanoid.c -o nanoid.o -DNDEBUG
In file included from nanoid.c:35:
./nanoid_rand.h:112:11: error: call to undeclared function 'getentropy'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
    ret = getentropy(buf, n);
          ^
1 error generated.

Error: Failed installing dependency: https://luarocks.org/lua-nanoid-1.1.0-1.src.rock - Build error: Failed compiling object nanoid.o
make: *** [deps] Error 1
```

**Comment by liweitianux** [comment]
Sigh, macOS is so weird. Fixed it in release v1.1.1. Please retry.
